src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/TransactionType.java

```diff
@@ -0,0 +1,25 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import java.util.Arrays;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+public interface TransactionType {
+
+    static TransactionType fromCode(String code) {
+        return Stream.of(
+                Arrays.stream(Inbound.values()),
+                Arrays.stream(Outbound.values()))
+                .flatMap(Function.identity())
+                .map(TransactionType.class::cast)
```

Comment:
`Inbound` and `Outbound` are already `TransationType` extensions, so the cast should be unnecessary